### DISCUSSION TITLE
Razor-like Templating Engine for UI

### DISCUSSION THREAD

**Discussion by kumja1** [CosmosOS/Cosmos#3097] [General]
This issue is more of a discussion on how a razor-like templating engine could be implemented in cosmos, which makes it easier for newcomers to design the UI of their os. I am aware that cosmos currently lacks multi-threading, but as mentioned earlier this is more of a way to get an idea of how such a feature might look like.

**Comment by kumja1** [comment]
@zarlo I see you have attempted something like this before with Kanga. Based on your experience, which would you recommend for cosmos, a widget-based framework like flutter or a template language like razor

**Reply by zarlo** [reply]
like it can be done, it would live never be build in to cosmos as that is out of scope of the project
